faunadb-java/src/main/java/com/faunadb/client/types/time/HighPrecisionTime.java

```diff
@@ -59,52 +64,58 @@ private static int toInt(String value) {
    * @param microsToAdd microseconds to add to the initial timestamp
    * @param nanosToAdd  nanoseconds to add to the initial timestamp
    */
-  public HighPrecisionTime(Instant initialTime, int microsToAdd, int nanosToAdd) {
+  public static HighPrecisionTime fromInstant(Instant initialTime, long microsToAdd, long nanosToAdd) {
```

Comment:
This needs some overloads in java if we expect users to use this. The example usage in tests is a bit cumbersome.

I'd suggest a public interface of `fromInstant(Instant inst)` and `fromInstant(Instant int, long nanos)`